clients/go/pkg/commands/stream_jobs_test.go

```diff
@@ -0,0 +1,198 @@
+		ElementInstanceKey:       123,
+		ProcessDefinitionKey:     124,
+		BpmnProcessId:            "fooProcess",
+		ProcessInstanceKey:       1233,
+		ElementId:                "foobar",
+		ProcessDefinitionVersion: 12345,
+		CustomHeaders:            "{\"foo\": \"bar\"}",
+		Variables:                "{\"foo\": \"bar\"}",
+	}
+	job2 := pb.ActivatedJob{
+		Key:           123,
+		Type:          "foo",
+		Retries:       3,
+		Deadline:      123123,
+		Worker:        DefaultJobWorkerName,
+		CustomHeaders: "{}",
+		Variables:     "{}",
+	}
+
+	expectedJobs := []entities.Job{{ActivatedJob: &job1}, {ActivatedJob: &job2}}
+
+	gomock.InOrder(
+		stream.EXPECT().Recv().Return(&job1, nil),
+		stream.EXPECT().Recv().Return(&job2, nil),
+		stream.EXPECT().Recv().Return(nil, io.EOF),
+	)
+
+	client.EXPECT().StreamActivatedJobs(gomock.Any(), &utils.RPCTestMsg{Msg: request}).Return(stream, nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), utils.DefaultTestTimeout)
+	defer cancel()
+
+	jobsChan := make(chan entities.Job, 2)
+	defer close(jobsChan)
+	err := NewStreamJobsCommand(client, func(context.Context, error) bool {
+		return false
+	}).JobType("foo").Consumer(jobsChan).Send(ctx)
+
+	assert.NoError(t, err)
+
+	// simulate job receive
+	jobs := []entities.Job{<-jobsChan, <-jobsChan}
+
+	if len(jobs) != len(expectedJobs) {
+		t.Error("Failed to receive all jobs: ", jobs, expectedJobs)
+	}
+
+	for i, job := range jobs {
+		if !reflect.DeepEqual(job, expectedJobs[i]) {
+			t.Error("Failed to receive job: ", job, expectedJobs[i])
+		}
+	}
+
+}
+
+func TestStreamJobsCommandWithTimeout(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	client := mock_pb.NewMockGatewayClient(ctrl)
+	stream := mock_pb.NewMockGateway_StreamActivatedJobsClient(ctrl)
+
+	request := &pb.StreamActivatedJobsRequest{
+		Type:    "foo",
+		Timeout: 60 * 1000,
+		Worker:  DefaultJobWorkerName,
+	}
+
+	stream.EXPECT().Recv().Return(nil, io.EOF)
+	client.EXPECT().StreamActivatedJobs(gomock.Any(), &utils.RPCTestMsg{Msg: request}).Return(stream, nil)
+
+	ctx, cancel := context.WithTimeout(context.Background(), utils.DefaultTestTimeout)
+	defer cancel()
+
+	jobsChan := make(chan entities.Job, 5)
+	defer close(jobsChan)
+	err := NewStreamJobsCommand(client, func(context.Context, error) bool {
+		return false
+	}).JobType("foo").Consumer(jobsChan).Timeout(1 * time.Minute).Send(ctx)
+
+	if err != nil {
+		assert.NoError(t, err, "Failed to send request")
+	}
+}
+
+func TestStreamJobsCommandWithWorkerName(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	client := mock_pb.NewMockGatewayClient(ctrl)
+	stream := mock_pb.NewMockGateway_StreamActivatedJobsClient(ctrl)
+
+	fetchVariables := []string{"foo", "bar", "baz"}
+	request := &pb.StreamActivatedJobsRequest{
+		Type:          "foo",
+		Timeout:       300 * 1000,
+		Worker:        DefaultJobWorkerName,
+		FetchVariable: fetchVariables,
+	}
```

Comment:
The different tests seem to test the different parameters in this request, is this correct? Wondering whether we can highlight or better assert this?